python/hail/api2/matrixtable.py

```diff
@@ -2538,5 +2538,36 @@ def from_rows_table(cls, table):
         jmt = scala_object(Env.hail().variant, 'MatrixTable').fromRowsTable(table._jt)
         return MatrixTable(jmt)
 
+    @handle_py4j
+    @typecheck_method(p=numeric,
+                      seed=integral)
+    def sample_rows(self, p, seed=0):
+        """Downsample the dataset by keeping each row with probability ``p``.
+
+        Examples
+        --------
+
+        Downsample the dataset to approximately 1% of its rows.
+
+        >>> small_dataset = dataset.sample_rows(0.01)
+
+        Parameters
+        ----------
+        p : :obj:`float`
+            Probability of keeping each row.
+        seed : :obj:`int`
+            Random seed.
+
+        Returns
+        -------
+        :class:`.MatrixTable`
+            Matrix with approximately ``p * num_rows`` rows.
```

Comment:
I prefer `Dataset` since the class is next door. You could also say `MatrixTable` I guess. But `Matrix` is going to be ambiguous with numeric matrices.

Reply:
I've bene writing "Matrix table".

Reply:
but @tpoterba might have opinions.

Reply:
I dont care, I still don't particularly like MatrixTable  but it's the best we've got

Reply:
yeah, "Matrix table" seems most consistent. I'll use that going forward

@tpoterba MatrixTable is growing on me. Later we can have TensorTable.